lib/src/charts/painters/cartesian/bar_painter.dart

```diff
@@ -24,17 +23,83 @@ class BarPainter implements CartesianPainter {
   late Paint _barPaint;
   late Paint _barStroke;
 
-  BarPainter({
-    required this.useGraphUnits,
-  }) {
+  BarPainter({required this.useGraphUnits}) {
     _barPaint = Paint();
     _barStroke = Paint()..style = PaintingStyle.stroke;
   }
 
+  _BarInteractionData _findNearestBar(
+    Offset position,
+    List<_BarInteractionData> bars, {
+    double fuzzinessRange = 0.0,
+  }) {
+    return bars.reduce((previous, current) {
+      final distToPrev = position.dx - previous.rect.right;
+      final distToCurrent = current.rect.left - position.dx;
+
+      return distToPrev >= distToCurrent ? current : previous;
+    });
+  }
+
+  @override
+  BarInteractionResult? hitTest(
```

Comment:
This needs to be optimized as we discussed. We should try to exit as early as possible.